tasks/test_e2e.yml

```diff
@@ -1,39 +1,80 @@
-    - name: "Extract URL from stf-connectors template"
-      shell: grep interconnect /home/stack/virt/stf-connectors* | sed 's;.*\(-service-telemetry.*\);default-prometheus-proxy\1;'
-      register: prom_url
-      failed_when: "prom_url.stdout_lines|length == 0"
+    - name: Get the default-prometheus-htpasswd secret
+      shell: 
+        cmd: |
+          oc  get secret default-prometheus-htpasswd -ojson | jq '.data.auth, .data.password' |  sed 's/"//g'
+      register: prom_secret
     - debug:
-        var: prom_url.stdout_lines
+        var: prom_secret | string
+
+    - name: "Get the prom URL"
+      shell:
+        cmd: |
+          oc get route/default-prometheus-proxy -ojson | jq '.spec.host' | sed 's/"//g'
+      register: prom_route
 
     - debug:
-        msg: The URL is {{ prom_url.stdout_lines }}
+        var:  prom_route.stdout
+
+    - name: "Get the prom creds from the secret"
+      set_fact:
+        prom_user: "{{ ( prom_secret.stdout_lines[0] | b64decode | split(':'))[0] }}"
+        prom_pass: "{{ prom_secret.stdout_lines[1] | b64decode }}"
+        prom_url: "{{ prom_route.stdout }}"
+    - debug: var=prom_user
+
 
+ 
     - name: "RHELOSP-37759"
-      shell: /usr/bin/curl -k -u "internal:1tp9zXur-t8Ve,wPkVgE" -g https://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_cpu_percent {plugin_instance="0"}[1m]' > /tmp/query_collectd_cpu_percent
+      shell:
+        cmd: >-
+          /usr/bin/curl -k -u "{{ prom_user }}:{{ prom_pass }}" \
+            -g https://{{ prom_url }}/api/v1/query? \
+             --data-urlencode 'query=collectd_cpu_percent {plugin_instance="0"}[1m]' \
+             --output /tmp/query_collectd_cpu_percent
       register: checkmyconf
       failed_when:
         - checkmyconf.rc !=0
 
     - name: "RHELOSP-57528"
-      shell: /usr/bin/curl -k -u "internal:1tp9zXur-t8Ve,wPkVgE" -g https://{{ prom_url.stdout_lines[0] }}/api/v1/query? --data-urlencode 'query=collectd_ceph_ceph_bytes {plugin_instance="ceph-osd.5"}[1m]' > /tmp/query_ceph_ceph_bytes
+      shell:
+        cmd: >-
+          /usr/bin/curl -k -u "{{ prom_user }}:{{ prom_pass }}" \
+            -g https://{{ prom_url }}/api/v1/query? \
```

Comment:
Can you update the indentation so these lines are aligned? 

First line is okay, and each subsequent line should be at the same indentation as the second one.
Same comment for the other shell tasks.

e.g. 
```
  /usr/bin/curl -k -u "{{ prom_user }}:{{ prom_pass }}" \
    -g https://{{ prom_url }}/api/v1/query? \
    --data-urlencode 'query=collectd_ceph_ceph_bytes {plugin_instance="ceph-osd.5"}[1m]' \
    --output /tmp/query_ceph_ceph_bytes
```